accepted/00000-replace-salt-toaster.md

```diff
@@ -0,0 +1,192 @@
+plan to do that. While we can't use sumaform to provision GitHub-hosted runners,
+we can prepare the runner before executing a test suite run. With Salt Toaster,
+we ran the tests in a Docker container based on openSUSE Leap 15.3.
+GitHub-hosted runners only use one Linux distribution, Ubuntu. At the time of
+writing, only `ubuntu-22.04` ships Salt, which we will need to apply the
+provisioning state.
+
+The GitHub Action workflow definitions in openSUSE/salt are extended with a
+build job that installs `salt-common`, checks out the salt-jenkins repository,
+and applies the `golden-images-provision.sls` with `salt-call`.
+
+Applying the provisioning state in GitHub runners takes roughly 60 to 90
+seconds, while `nox` regularly needs more than that for setting up a virtual
+environment to run the tests in. We reduce the setup time, with the
+[cache](https://github.com/actions/cache/blob/main/examples.md#simple-example)
+action.
+
+## Terracumber
+
+Terracumber is a tool used by our QE squad to operate terraform and cucumber for
+their regular test suite execution. The same tool can be extended to set up the
+required VMs and start the Salt test suite execution.
+
+## Executing the Salt Test Suite
+
+The test suite uses `nox` to set up a Python virtual environment with all
+required dependencies. Arbitrary CLI flags for `pytest`, the used Python testing
+framework, can be passed through. We use this for two purposes: passing the path
+to exactly those test groups that we want to run, and to skip unstable tests.
+
+### Test Script
+
+A test script that passes the correct arguments makes it easy to run the same
+test selection in different environments: locally, in Jenkins or in GitHub
+Actions on Pull Requests. The script is simple on purpose. It takes a
+configuration file that maps test group names to the test locations and includes
+a skip list, one for each test group.
+
+Example:
+```sh
+$ cat test_config.yml
+groups:
+  unit:
+    tests_dirs:
+      - tests/pytests/unit
+      - tests/unit
+    skipped_tests:
+      - tests/unit/flaky.py::flaky_func
+$ test_script unit
+-> nox --session 'pytest-3(coverage=False)' -- --deselect=tests/unit/flaky.py::flaky_func tests/pytests/unit tests/unit
+```
+
+### Skipping Tests
+
+Causing a new test failure needs to feel bad. Turning a test result from green
+to red is a strong signal that either the product code or the test code has not
+reached an acceptable quality level yet. At the same time, we inherit flaky
+tests from upstream and don't have the time to fix them. These tests can be
+skipped to keep the overall test results green. But we need to be careful: when
+skipping tests is more convenient than fixing them, we risk skipping too
+many -- which directly lowers the value we get from running tests in the first
+place. Adding a test to the "skip list" is as big of a bad smell as is turning
+the result red.
+
+The skip list is maintained in a GitHub repository. All changes must be
```

Comment:
Does that mean a _different_ GitHub repository (than the main salt one)?

Reply:
Yes that is correct (don't have the link at hand sadly).

Reply:
Currently a similar list in defined in [Salt Toaster](https://github.com/openSUSE/salt-toaster/blob/master/conftest_source_nox.py#L1343).

I would put the skip list in the same repository as the above mentioned test Script, mainly to reduce the number of repositories used. Re-using `openSUSE/salt` only works if we disable direct pushes to the branches that contain the skip list and always remember to exclude all changes touching the test script and skip list from our packaging automation. IMO it makes sense to keep the script and the definition outside of the `openSUSE/salt` repo.